priv/repo/migrations/20230419134906_notification_users_index.exs

```diff
@@ -0,0 +1,7 @@
+defmodule Skate.Repo.Migrations.Notification_users_index do
+  use Ecto.Migration
+
+  def change do
+    create index("notifications_users", [:user_id])
```

Comment:
Ah sorry I forgot to respond to this! `notification_id` doesn't have an index on it right now. I'm not sure if it is as necessary at this point based on how frequently it is queried & the latency of those queries. It seems the only time we perform a read query using `notification_id` is in `Notification.update_read_states`. Looking at the top queries in over the last week in RDS console, I see updating `notifications_users` read state has an avg latency of 8.72ms, compared to the 30.81ms for selecting all notifications for a user. 

If it does seem worthwhile to add that index while we're at it, I think it would be worth doing so in a separate PR so that we can merge & deploy them separately in case the duration of running this migration results in issues.

Reply:
That point about not wanting to extend the duration of the migration is a good one. I'd say given what you've found about the currently-existing queries we can leave it for now and just keep it in mind if we add queries in the future that might benefit from it.